### PR TITLE
Create opensynth package

### PR DESCRIPTION
**Pull request by shengy90**
1. Create Opensynth package to deploy to Pypi
2. Changing project license from MIT to Apache2.0
3. Add shell scripts to check all .py files in `src` has required licence and copyright headers.
4. Create deployment workflow for releases.
5. Potential (explore later) - test releases to TestPyPI prior to creating a release and actually publishing to PyPI

### CODE REVIEW

File: src/opensynth/__init__.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Copyright Contributors to the Opensynth-energy Project.
+# SPDX-License-Identifier: Apache-2.0
+
+__version__ = "0.0.1"
```
Comment (shengy90) [inline src/opensynth/__init__.py]: Outdated - next commit fixes this to __version__="0.0.0-alpha".

When creating release branch that ultimately publish this to PyPI - we'll bump this up in the release PR.

File: .github/workflows/prod_deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+name: Deploy Package
+on:
+  release:
+    types: [published]
```
Comment (anguschadney) [inline .github/workflows/prod_deploy.yaml:4]: what happens if we need to unpublish a release? i.e. if we spot something dodgy...
do we just manually roll back on PyPI?

File: .github/workflows/test_and_lint.yaml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-name: Test and lint
+name: Deploy Package
```
Comment (anguschadney) [inline .github/workflows/test_and_lint.yaml:1]: What is this doing now? are we planning to run tests later?

**Review by anguschadney — COMMENTED** [review]
Looks great to me, couple of comments on the GH actions...

File: .github/workflows/prod_deploy.yaml (hunk 2 above)
Comment (shengy90) [reply .github/workflows/prod_deploy.yaml:4]: Can't roll back - once live on Pypi - you have to yank it. Then release a post-release:
https://snarky.ca/what-to-do-when-you-botch-a-release-on-pypi/

Unfortunately once live on PyPI, the same version number can't be used again. We'll have to suffix with `post1` etc. Hence I created a dummy github action for the release branch to test publishing to TestPYPI as a final 'line of defence' before pushing stuff to prod.

Still thinking about it, feels like an overkill at this stage, so not quite set in stone.

File: .github/workflows/test_and_lint.yaml (hunk 3 above)
Comment (shengy90) [reply .github/workflows/test_and_lint.yaml:1]: Copy-paste error - will fix!

Comment (anguschadney) [reply .github/workflows/test_and_lint.yaml]: cool but we still want to run pytest etc. don't we?

**Review by anguschadney — APPROVED** [review]
👍 let's go!

**Pull request merged**